### PR TITLE
ovl_Kamakura_Npc0 OK

### PR DESCRIPTION
**Pull request by MegaIDK**

### CODE REVIEW

File: src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c [hunk 1]
```diff
@@ -20,25 +24,57 @@ ActorProfile Kamakura_Npc0_Profile = {
...
     /* */ aKM0_actor_init,
-    /* */ (ActorFunc)none_proc1,
+    /* */ (void*)none_proc1,
     /* */ aKM0_actor_save,
 };
-#endif
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0/aKM0_actor_ct.s")
+struct_809AEFA4 D_809DF9B4_jp = {
+    func_809DF92C_jp, func_809DF958_jp, 3, func_809DF884_jp, func_809DF8B4_jp, func_809DF8E4_jp,
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c:32]:
```suggestion
    aKM0_actor_move, aKM0_actor_draw, 3, aKM0_tralk_request, aKM0_talk_init, aKM0_talk_end_chk,
```

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 2]
```diff
@@ -27,10 +27,7 @@ ActorProfile Npc_Engineer_Profile = {
 };
 
 struct_809AEFA4 D_809AEFA4_jp = {
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:29]: Same as before, this is `ct_data`

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 3]
```diff
@@ -27,10 +27,7 @@ ActorProfile Npc_Engineer_Profile = {
...
-    func_809AEF28_jp,
-    func_809AEF54_jp,
-    4,
-    0,
+    func_809AEF28_jp, func_809AEF54_jp, 4, 0, 0, 0,
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:30]:
```suggestion
    aNEG_actor_move, aNEG_actor_draw, 4, 0, 0, 0,
```

File: src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c [hunk 4]
```diff
@@ -20,25 +24,57 @@ ActorProfile Kamakura_Npc0_Profile = {
...
     /* */ aKM0_actor_dt,
     /* */ aKM0_actor_init,
-    /* */ (ActorFunc)none_proc1,
+    /* */ (void*)none_proc1,
     /* */ aKM0_actor_save,
 };
-#endif
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0/aKM0_actor_ct.s")
+struct_809AEFA4 D_809DF9B4_jp = {
```
[the remark this answers is not in the document]
Comment (MegaIDK) [reply src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c:31]: do i have to change the struct typedef too or just the name?
like `ct_data aKM0_ct_data` or somehting

[the remark this answers is not in the document]
Comment (Prakxo) [reply src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c:31]: I'm not sure if each NPC actor has a custom ct_data structure or if they inherit from the base npc one, so leaving it unnamed at the moment might be the best

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c (hunk 3 above)
Comment (MegaIDK) [reply src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:30]: oh also i hadnt noticed that idk maybe format squished the whole struct into a single line, is that ok?

Comment (emilybrooks) [reply src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c]: Its sort of up to whatever looks best. a lot of times you can play with adding or removing the final comma in the struct which causes the format script to do different things.

File: include/unknown_structs.h [hunk 5]
```diff
@@ -35,9 +35,11 @@ typedef struct Game_Play_unk_0110 {
...
     /* 0x0 */ UNK_PTR unk_0;
-    /* 0x0 */ UNK_PTR unk_4;
-    /* 0x0 */ UNK_TYPE unk_8;
-    /* 0x0 */ UNK_TYPE unk_C;
-} struct_809AEFA4; // size >= 0x10
+    /* 0x4 */ UNK_PTR unk_4;
+    /* 0x8 */ UNK_TYPE unk_8;
+    /* 0xC */ UNK_PTR unk_C;
+    /* 0x10 */ UNK_PTR unk_10;
+    /* 0x14 */ UNK_PTR unk_14;
```
Comment (AngheloAlf) [inline include/unknown_structs.h:42]: Same here
```suggestion
    /* 0x04 */ UNK_PTR unk_4;
    /* 0x08 */ UNK_TYPE unk_8;
    /* 0x0C */ UNK_PTR unk_C;
    /* 0x10 */ UNK_PTR unk_10;
    /* 0x14 */ UNK_PTR unk_14;
```

File: src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c [hunk 6]
```diff
@@ -20,25 +24,57 @@ ActorProfile Kamakura_Npc0_Profile = {
...
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0/aKM0_actor_dt.s")
+void aKM0_actor_init(Actor* thisx, Game_Play* game_play) {
+    common_data.unk_1004C->unk_CC(thisx, game_play);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0/aKM0_actor_init.s")
+void aKM0_talk_request(UNK_TYPE arg1, UNK_TYPE arg2 UNUSED) {
+    common_data.unk_10078->unk_00(arg1);
+}
```
[the remark this answers is not in the document]
Comment (MegaIDK) [reply src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c:56]: i think so yeah, gave me `(? arg1)` or something like that

**Review by hensldm — COMMENTED** [review]
1 nit

File: src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c [hunk 7]
```diff
@@ -20,25 +24,57 @@ ActorProfile Kamakura_Npc0_Profile = {
...
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0/func_809DF8B4_jp.s")
+s32 aKM0_talk_end_chk(UNK_TYPE arg0, UNK_TYPE arg1 UNUSED) {
+    s32 var_v1;
+    s32 sp1C;
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0/func_809DF8E4_jp.s")
+    sp1C = 0;
+    var_v1 = sp1C;
```
[the remark this answers is not in the document]
Comment (MegaIDK) [reply src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c:67]: oh yea i didnt really mess with any of that cus i didnt want to make it not match by accident or anything but like yea sp1C is kinda useless there

[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/actors/ovl_Kamakura_Npc0/ac_kamakura_npc0.c]: Sorry you should keep the variable and return type as `s32`, and just use `true/false`. Using `bool` doesn't work in all cases (sometimes you need things like s16 or something else) so we don't really like using it.

**Review by hensldm — APPROVED** [review]

**Pull request merged**